File: android/src/main/java/com/mobilecoin/mobilecoin_flutter/FfiTransferPayload.java
package com.mobilecoin.mobilecoin_flutter;

import com.mobilecoin.lib.RistrettoPublic;
import com.mobilecoin.lib.TransferPayload;

import androidx.annotation.Keep;

@Keep
public class FfiTransferPayload {

    private FfiTransferPayload() {
    }

    public static byte[] getBip39Entropy(int payloadId) {
        TransferPayload transferPayload = (TransferPayload) ObjectStorage.objectForKey(payloadId);
        return transferPayload.getBip39Entropy();
    }

    public static String getMemo(int payloadId) {
        TransferPayload transferPayload = (TransferPayload) ObjectStorage.objectForKey(payloadId);
        return transferPayload.getMemo();
    }

    public static int getPublicKey(int payloadId) {
        TransferPayload transferPayload = (TransferPayload) ObjectStorage.objectForKey(payloadId);
        RistrettoPublic publicKey = transferPayload.getPublicKey();
        final int hashCode = publicKey.hashCode();
        ObjectStorage.addObject(hashCode, publicKey);
        return hashCode;
    }
}

File: ios/Classes/FfiTransferPayload.swift
// Copyright (c) 2021-2024 MobileCoin. All rights reserved.

import Foundation
import MobileCoin

struct FfiTransferPayload {
    
    struct GetBip39Entropy: Command {
        func execute(args: [String : Any], result: @escaping FlutterResult) throws {
            guard let payloadId: Int = args["id"] as? Int,
                  let transferPayload = ObjectStorage.objectForKey(payloadId) as? TransferPayload else {
                      throw PluginError.invalidArguments
                  }
            result(transferPayload.bip39)
        }
    }
    
    struct GetMemo: Command {
        func execute(args: [String : Any], result: @escaping FlutterResult) throws {
            guard let payloadId: Int = args["id"] as? Int,
                  let transferPayload = ObjectStorage.objectForKey(payloadId) as? TransferPayload else {
                      throw PluginError.invalidArguments
                  }
            result(transferPayload.memo)
        }
    }
    
    struct GetPublicKey: Command {
        func execute(args: [String : Any], result: @escaping FlutterResult) throws {
            guard let payloadId: Int = args["id"] as? Int,
                  let transferPayload = ObjectStorage.objectForKey(payloadId) as? TransferPayload else {
                      throw PluginError.invalidArguments
                  }
            let hashCode = transferPayload.hashValue
            result(hashCode)
        }
    }
}
